### PR TITLE
Changes to secure time (addresses Issue #169)

### PR DESCRIPTION
**Pull request by jhdalek55**
Per @iramcdonald to resolve issue #169

### CODE REVIEW

**Comment by jhdalek55** [comment]
@allen-cain-tm @trishankatdatadog  or others, please review. These were changes suggested by @iramcdonald

**Comment by jhdalek55** [comment]
Per the consensus of our Standards meeting on 12/7, we are modifying this to a SHOULD. The PR has been updated accordingly.

File: uptane-standard.md [hunk 1]
```diff
@@ -707,7 +707,7 @@ There may be several different filenames that all refer to the same image binary
 
 #### Send latest time to Secondaries {#send_time_primary}
 
-Unless the Secondary ECU has its own way of verifying the time or does not have the capacity to verify a time message, the Primary is CONDITIONALLY REQUIRED to send the time to each ECU. The Secondary will verify the time message, then overwrite its current time with the received time.
+The Primary SHOULD send the time to each ECU.
```
Comment (pattivacek) [inline uptane-standard.md:710]: I feel like we're missing something by entirely removing the option for a Secondary to have it's own source of secure time, but maybe that's unrealistic anyway. I'll follow the insight of others.

**Comment by iramcdonald** [comment]
Hi Patti,

In yesterday's discussion (but not yet the Deployment BP) we agreed the
SHOULD for the Primary sending time messages is intentionally silent
about any other time sources (e.g., internal realtime clock in a Telematics
box). And the Secondary can also choose to use the time message from
the Primary or use another time source internal to the car.

The general assumption is that a Secondary is not and *should not* be on
the public Internet, so its time sources are internal to the vehicle.

Cheers,
- Ira

*Ira McDonald (Musician / Software Architect)*

*Chair - SAE Trust Anchors and Authentication TF*
*Co-Chair - TCG Trusted Mobility Solutions WG*

*Co-Chair - TCG Metadata Access Protocol SG*








*Chair - Linux Foundation Open Printing WGSecretary - IEEE-ISTO Printer
Working GroupCo-Chair - IEEE-ISTO PWG Internet Printing Protocol WGIETF
Designated Expert - IPP & Printer MIBBlue Roof Music / High North
Inchttp://sites.google.com/site/blueroofmusic
<http://sites.google.com/site/blueroofmusic>http://sites.google.com/site/highnorthinc
<http://sites.google.com/site/highnorthinc>mailto: ***@***.***
***@***.***>(permanent) PO Box 221  Grand Marais, MI 49839
906-494-2434*

**Review by pattivacek — COMMENTED** [review]
> the SHOULD for the Primary sending time messages is intentionally silent about any other time sources (e.g., internal realtime clock in a Telematics box). And the Secondary can also choose to use the time message from the Primary or use another time source internal to the car.

This sounds fine, but then it's a bit strange that we say you should sent time from the Primary to a Secondary even if the Secondary is going to ignore it and use something else. I don't see the problem with the original text in the "Send latest time to Secondaries" section. The bit about overwriting is probably unnecessary overprescriptive, but the first sentence seems fine.

I'll defer to y'all, though. This really isn't my area of expertise.

Removing the rest of the Time Server references looks fine to me.

**Comment by iramcdonald** [comment]
H Patti,

The reason behind the *plain* statement that the Primary SHOULD send time
messages to its
Secondaries is KISS.  A Primary inherently must have Internet connectivity,
but a Secondary
should not have Internet connectivity.  There are multiple possible time
sources within a vehicle.
They probably vary in reliability and frequency of update.

The pretty serious problem with the original sentence was that it required
the Primary to have
out-of-band configuration knowledge of the construction of the Secondaries
(i.e., whether they
had any other time source at all).  Out-of-band knowledge is bad protocol
design and leads to
more complicated and brittle implementations.

I hope this helps to illuminate my thinking.

Cheers,
- Ira


*Ira McDonald (Musician / Software Architect)*

*Chair - SAE Trust Anchors and Authentication TF*
*Co-Chair - TCG Trusted Mobility Solutions WG*

*Co-Chair - TCG Metadata Access Protocol SG*








*Chair - Linux Foundation Open Printing WGSecretary - IEEE-ISTO Printer
Working GroupCo-Chair - IEEE-ISTO PWG Internet Printing Protocol WGIETF
Designated Expert - IPP & Printer MIBBlue Roof Music / High North
Inchttp://sites.google.com/site/blueroofmusic
<http://sites.google.com/site/blueroofmusic>http://sites.google.com/site/highnorthinc
<http://sites.google.com/site/highnorthinc>mailto: ***@***.***
***@***.***>(permanent) PO Box 221  Grand Marais, MI 49839
906-494-2434*

**Comment by pattivacek** [comment]
Thanks, Ira, that does make sense. The default choice for Secondaries is probably usually going to be to trust the Primary's time, and anything else is likely to be exceptional, which the SHOULD allows for. There shouldn't be a problem there.

**Review by pattivacek — APPROVED** [review]

**Review by trishankatdatadog — APPROVED** [review]

**Pull request merged**